# Heart Disease: Preprocessing and Baseline Model

In this notebook we prepare the data for modelling and train a logistic regression baseline. Everything we learned in the EDA notebook shapes the choices here: impute the few missing values, one-hot encode categorical columns, scale numeric columns, and drop `num` so the model never sees the answer.

In [1]:
import pandas as pd

df = pd.read_csv("../data/heart_raw.csv")
df["target"] = (df["num"] > 0).astype(int)
df = df.drop(columns=["num"])

print(df.shape)
df.head()

(303, 14)


,age,sex,cp,trestbps,chol,fbs,restecg,thalach,exang,oldpeak,slope,ca,thal,target
0,63,1,1,145,233,1,2,150,0,2.3,3,0.0,6.0,0
1,67,1,4,160,286,0,2,108,1,1.5,2,3.0,3.0,1
2,67,1,4,120,229,0,2,129,1,2.6,2,2.0,7.0,1
3,37,1,3,130,250,0,0,187,0,3.5,3,0.0,3.0,0
4,41,0,2,130,204,0,2,172,0,1.4,1,0.0,3.0,0


## 1. Train/test split

We hold back 20% of patients as a test set. It stays untouched until we have chosen our final model, so it gives an honest estimate of performance on unseen patients. We stratify on the target so both sets keep the same disease/no-disease ratio, which matters with only 303 rows.

In [2]:
from sklearn.model_selection import train_test_split

X = df.drop(columns=["target"])
y = df["target"]

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

print("Train:", X_train.shape, " Test:", X_test.shape)
print("Disease rate, train:", round(y_train.mean(), 3))
print("Disease rate, test: ", round(y_test.mean(), 3))

Train: (242, 13)  Test: (61, 13)
Disease rate, train: 0.459
Disease rate, test:  0.459
